File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/utils/event/click/ClickEventAction.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.utils.event.click;


import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.EnumWrapper;

@WrapperClass(mcpName = "net.minecraft.event.ClickEvent$Action",targetMap = MapEnum.MDK189)
public class ClickEventAction extends EnumWrapper {
    @WrapEnum(mcpName = "OPEN_URL",targetMap = MapEnum.MDK189)
    public static Enum OPEN_URL;
    @WrapEnum(mcpName = "OPEN_FILE",targetMap = MapEnum.MDK189)
    public static Enum OPEN_FILE;
    @WrapEnum(mcpName = "RUN_COMMAND",targetMap = MapEnum.MDK189)
    public static Enum RUN_COMMAND;
    @WrapEnum(mcpName = "TWITCH_USER_INFO",targetMap = MapEnum.MDK189)
    public static Enum TWITCH_USER_INFO;
    @WrapEnum(mcpName = "SUGGEST_COMMAND",targetMap = MapEnum.MDK189)
    public static Enum SUGGEST_COMMAND;
    @WrapEnum(mcpName = "CHANGE_PAGE",targetMap = MapEnum.MDK189)
    public static Enum CHANGE_PAGE;
    public ClickEventAction(Object obj) {
        super(obj);
    }
}

File: src/main/kotlin/rbq/lycoris/client/wrapper/wrappers/util/event/click/ClickEventAction.kt
package rbq.lycoris.client.wrapper.wrappers.util.event.click

import rbq.lycoris.client.wrapper.MapEnum
import rbq.lycoris.client.wrapper.annotation.WrapEnum
import rbq.lycoris.client.wrapper.annotation.WrapperClass
import rbq.lycoris.client.wrapper.wrappers.EnumWrapper

@WrapperClass(mcpName = "net.minecraft.event.ClickEvent\$Action", targetMap = MapEnum.VANILLA189)
class ClickEventAction(obj: Any) : EnumWrapper(obj) {
    companion object {
        @WrapEnum(mcpName = "OPEN_URL", targetMap = MapEnum.VANILLA189)
        lateinit var OPEN_URL: Enum<*>

        @WrapEnum(mcpName = "OPEN_FILE", targetMap = MapEnum.VANILLA189)
        lateinit var OPEN_FILE: Enum<*>

        @WrapEnum(mcpName = "RUN_COMMAND", targetMap = MapEnum.VANILLA189)
        lateinit var RUN_COMMAND: Enum<*>

        @WrapEnum(mcpName = "TWITCH_USER_INFO", targetMap = MapEnum.VANILLA189)
        lateinit var TWITCH_USER_INFO: Enum<*>

        @WrapEnum(mcpName = "SUGGEST_COMMAND", targetMap = MapEnum.VANILLA189)
        lateinit var SUGGEST_COMMAND: Enum<*>

        @WrapEnum(mcpName = "CHANGE_PAGE", targetMap = MapEnum.VANILLA189)
        lateinit var CHANGE_PAGE: Enum<*>
    }
}